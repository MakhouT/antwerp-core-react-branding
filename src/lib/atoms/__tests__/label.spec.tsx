import * as React from "react";
import * as renderer from "react-test-renderer";

import * as A from "../..";

describe("Label", () => {
    it("Renders default, string", () => {
        // Act
        const component: any = renderer.create(
            <A.Label
                value="value"
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders default, number", () => {
        // Act
        const component: any = renderer.create(
            <A.Label
                value={10}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders Label with level danger", () => {
        // Arrang
        const level = A.Levels.Danger;

        // Act
        const component: any = renderer.create(
            <A.Label
                value={10}
                level={level}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree.props.className).toContain(`a-label--${level}`);
        expect(tree).toMatchSnapshot();
    });

    it("Renders Label with level warning", () => {
        // Arrang
        const level = A.Levels.Warning;

        // Act
        const component: any = renderer.create(
            <A.Label
                value={10}
                level={level}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree.props.className).toContain(`a-label--${level}`);
        expect(tree).toMatchSnapshot();
    });
});
