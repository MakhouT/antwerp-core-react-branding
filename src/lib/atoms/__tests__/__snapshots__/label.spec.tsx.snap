// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Renders Label with level danger 1`] = `
<span
  className="a-label a-label--danger"
>
  10
</span>
`;

exports[`Label Renders Label with level warning 1`] = `
<span
  className="a-label a-label--warning"
>
  10
</span>
`;

exports[`Label Renders default, number 1`] = `
<span
  className="a-label a-label--"
>
  10
</span>
`;

exports[`Label Renders default, string 1`] = `
<span
  className="a-label a-label--"
>
  value
</span>
`;
